### PR TITLE
exif: drop the 6-byte Exif JPEG APP1 header

### PR DESCRIPTION
**Pull request by kmilos**
The 6-byte header is only specified in the Exif spec for storing in JPEG APP1

### CODE REVIEW

**Comment by kmilos** [comment]
See also https://gitlab.gnome.org/GNOME/gimp/-/blob/5ea200f67a4c3771b74f250698d9995d3fdf2b3a/plug-ins/file-exr/openexr-wrapper.cc#L354

**Comment by kmilos** [comment]
We might want to double-check other export formats as well, or flip this around if necessary: make `dt_exif_read_blob()` output the blob without the header, and only add it where necessary.

**Comment by kmilos** [comment]
PNG also suffered the same problem, the others seem to be ok as they use `dt_exif_write_blob()` which takes cares of this and writes via exiv2.

@TurboGit I consider these a bug, would be nice to shoe it in for 4.0

TODO: switch PNG to eventually write `eXIf` dedicated chunk instead (`dt_exif_write_blob()` and exiv2 wouldn't work since it also writes to "Raw profile data exif")

**Comment by kmilos** [comment]
Finally I decided to drop this 6-byte altogether, exiv2 will do the correct thing when writing to JPEG anyway, and it's just confusing things for every other client.

**Comment by TurboGit** [comment]
@kmilos : Too late for 4.0 as the tag has been put yesterday night and a tarball for the release generated and sent to the packagers.

I have tagged that for 4.0.1. Thanks.

File: src/imageio/format/png.c [hunk 1]
```diff
@@ -52,7 +52,7 @@ typedef struct dt_imageio_png_gui_t
   GtkWidget *compression;
 } dt_imageio_png_gui_t;
 
-/* Write EXIF data to PNG file.
+/* Write EXIF data to PNG file (legacy, before PNG extension for eXIf chunk).
```
Comment (kmilos) [inline src/imageio/format/png.c:55]: [Exiftool actually uses this one now as well](https://exiftool.org/TagNames/PNG.html):

"this is where ExifTool will create new EXIF"

Comment (kmilos) [reply src/imageio/format/png.c]: Removed this commit as it lead to eXIf being written twice (and Xmp writing by exiv2 failing), so will follow up separately.

**Pull request merged**